expression/builtin_cast_test.go

```diff
@@ -212,6 +212,26 @@ func (s *testEvaluatorSuite) TestCast(c *C) {
 	c.Assert(terror.ErrorEqual(types.ErrOverflow, lastWarn.Err), IsTrue, Commentf("err %v", lastWarn.Err))
 	sc = origSc
 
+	// create table tt(a bigint unsigned);
+	// insert into tt values(18446744073709551615);
+	// select cast(a as decimal(65, 0)) from tt;
+	ft = &types.FieldType{
+		Tp:      mysql.TypeNewDecimal,
+		Flag:    mysql.BinaryFlag,
+		Charset: charset.CharsetBin,
+		Collate: charset.CollationBin,
+		Flen:    65,
+		Decimal: 0,
+	}
+	rt := types.NewFieldType(mysql.TypeLonglong)
+	rt.Flag = mysql.BinaryFlag | mysql.UnsignedFlag
+	f = BuildCastFunction(ctx, &Constant{Value: types.NewUintDatum(18446744073709551615), RetType: rt}, ft)
+	res, err = f.Eval(chunk.Row{})
+	c.Assert(err, IsNil)
+	u, err := res.GetMysqlDecimal().ToUint()
```

Comment:
It's not like `cast(cast(arg as decimal) as unsigned)` , because when cast(arg as decimal) returns signed， `res.GetMysqlDecimal().ToUint()` will return range exceed error.